apps/prairielearn/src/lib/course.ts

```diff
@@ -0,0 +1,231 @@
+import { callbackify, promisify } from 'util';
+import { exec } from 'child_process';
+import * as fs from 'fs-extra';
+import * as namedLocks from '@prairielearn/named-locks';
+import * as sqldb from '@prairielearn/postgres';
+import * as error from '@prairielearn/error';
+
+import { createServerJob } from './server-jobs';
+import { config } from './config';
+import * as chunks from './chunks';
+import { syncDiskToSqlWithLock } from '../sync/syncFromDisk';
+import { Course, CourseSchema, IdSchema, User, UserSchema } from './db-types';
+
+const sql = sqldb.loadSqlEquiv(__filename);
+
+/**
+ * Check that an assessment_instance_id really belongs to the given course_instance_id
+ *
+ * @param assessment_instance_id - The assessment instance to check.
+ * @param course_instance_id - The course instance it should belong to.
+ */
+export async function checkBelongsAsync(
+  assessment_instance_id: string,
+  course_instance_id: string,
+): Promise<void> {
+  if (
+    (await sqldb.queryOptionalRow(
+      sql.check_belongs,
+      { assessment_instance_id, course_instance_id },
+      IdSchema,
+    )) == null
+  ) {
+    throw new Error('access denied');
+  }
+}
+export const checkBelongs = callbackify(checkBelongsAsync);
+
+/**
+ * Return the name and UID (email) for every owner of the specified course.
+ *
+ * @param course_id The ID of the course.
+ */
+export async function getCourseOwners(course_id: string): Promise<User[]> {
+  return await sqldb.queryRows(sql.select_owners, { course_id }, UserSchema);
+}
+
+export function getLockNameForCoursePath(coursePath: string): string {
+  return `coursedir:${coursePath}`;
+}
+
+export async function selectCourseById(course_id: string): Promise<Course> {
+  return await sqldb.queryRow(sql.get_course_data, { course_id }, CourseSchema);
+}
```

Comment:
This is a duplicate of the new `selectCourseById` in `models/course.ts`. AFAICT this is a brand-new function you just added, so it should be a low lift to consolidate on the other one.

Some of these other functions might better belong in that other file, but I won't request a change here.

Reply:
`selectCourseById` done in 86519a13e3e92b2f54691ad39a183b9bc476978c. I considered moving `getCourseOwners` as well, but there are several references to it that I'd prefer not to touch in this PR.